modules/parser/src/synchronizers/xviz-stream-buffer.js

```diff
@@ -225,10 +229,16 @@ export default class XVIZStreamBuffer {
     const timesliceAtInsertPosition = timeslices[insertPosition];
 
     if (timesliceAtInsertPosition && timesliceAtInsertPosition.timestamp === timestamp) {
-      // Same timestamp, needs a merge
-      this._mergeTimesliceAt(insertPosition, timeslice);
+      // Same timestamp
+      if (updateType === 'INCREMENTAL') {
+        // Merge if it's an incremental update
+        this._mergeTimesliceAt(insertPosition, timeslice);
+      } else {
+        // Replace if it's a complete state
```

Comment:
we should check against "COMPLETE" specifically.  That way if we added a new update_type, 2.0 code running against 2.1 data should still "work" if not 100%.   It's unclear if the `updateType` other than incremental & complete are filtered out before we get here, that would have the same effect.